lib/matplotlib/_type1font.py

```diff
@@ -768,13 +861,262 @@ def transform(self, effects):
+        """Run one step in the charstring interpreter."""
+        if isinstance(opcode, int):
+            return set(), set(), buildchar_stack + [opcode], postscript_stack
+        elif opcode in {
+                'hsbw', 'sbw', 'closepath', 'hlineto', 'hmoveto', 'hcurveto',
+                'hvcurveto', 'rlineto', 'rmoveto', 'rrcurveto', 'vhcurveto',
+                'vlineto', 'vmoveto', 'dotsection', 'hstem', 'hstem3', 'vstem',
+                'vstem3', 'setcurrentpoint'
+        }:
+            return set(), set(), [], postscript_stack
+        elif opcode == 'seac':
+            codes = buildchar_stack[3:5]
+            glyphs: set[str] = {self.prop['Encoding'][x] for x in codes}
+            return glyphs, set(), [], postscript_stack
+        elif opcode == 'div':
+            num1, num2 = buildchar_stack[-2:]
+            return (
+                set(),
+                set(),
+                buildchar_stack[-2:] + [num1/num2], postscript_stack
+            )
+        elif opcode == 'callothersubr':
+            othersubr = buildchar_stack[-1]
+            n = buildchar_stack[-2]
+            if not isinstance(n, int):
+                _log.warning(
+                    f"callothersubr {othersubr} with non-integer argument count in "
+                    f"font {self.prop['FontName']}"
+                )
+                n = int(n)
+            args = buildchar_stack[-2-n:-2]
+            if othersubr == 3:  # Section 8.1 in Type-1 spec
+                postscript_stack.append(args[0])
+            else:
+                postscript_stack.extend(args[::-1])
+            return set(), set(), buildchar_stack[:-n-2], postscript_stack
+        elif opcode == 'callsubr':
+            subr = buildchar_stack[-1]
+            if not isinstance(subr, int):
+                _log.warning(
+                    f"callsubr with non-integer argument {subr} in font "
+                    f"{self.prop['FontName']}"
+                )
+                subr = int(subr)
+            glyphs, subrs, new_bc_stack, new_ps_stack = \
+                self._simulate(subr, buildchar_stack[:-1], postscript_stack)
+            return set(), subrs | {subr}, new_bc_stack, new_ps_stack
+        elif opcode == 'pop':
+            return (
+                set(),
+                set(),
+                buildchar_stack + [postscript_stack[-1]], postscript_stack[:-1]
+            )
+        else:
+            raise RuntimeError(f'opcode {opcode}')
+
+    def _subset_encoding(self, encoding: dict[int, str]) -> str:
+        """Return a PostScript encoding array for the encoding."""
+        result = [
+            '/Encoding 256 array\n0 1 255 { 1 index exch /.notdef put} for'
+        ]
+        result.extend(
+            f'dup {i} /{glyph} put'
+            for i, glyph in sorted(encoding.items())
+            if glyph != '.notdef'
+        )
+        result.append('readonly def\n')
+        return '\n'.join(result)
```

Comment:
Probably can join these:
```suggestion
        return '\n'.join([
            '/Encoding 256 array\n0 1 255 { 1 index exch /.notdef put} for',
            *(
                f'dup {i} /{glyph} put'
                for i, glyph in sorted(encoding.items())
                if glyph != '.notdef'
            ),
            'readonly def\n',
        ])
```
Maybe the same for the rest.

Reply:
Thanks!